src/main/java/com/truelayer/java/auth/entities/GenerateOauthTokenRequest.java

```diff
@@ -0,0 +1,45 @@
+package com.truelayer.java.auth.entities;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonValue;
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.databind.JsonSerializer;
+import com.fasterxml.jackson.databind.SerializerProvider;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import java.io.IOException;
+import java.util.List;
+import lombok.*;
+
+@Builder
+@Getter
+@ToString
+@EqualsAndHashCode
+public class GenerateOauthTokenRequest {
+    private String clientId;
+
+    private String clientSecret;
+
+    private final GrantType grantType = GrantType.CLIENT_CREDENTIALS;
+
+    @Singular
+    @JsonSerialize(using = ScopeSerializer.class)
+    @JsonProperty("scope")
+    private List<String> scopes;
+
+    @RequiredArgsConstructor
+    @Getter
+    public enum GrantType {
+        CLIENT_CREDENTIALS("client_credentials");
+
+        @JsonValue
+        private final String type;
+    }
+
+    public static class ScopeSerializer extends JsonSerializer<List<String>> {
```

Comment:
This is to serialize the scopes list into a whitespace separated string, like `payments signupplus`